### DISCUSSION TITLE
谈谈 Kubernetes 的问题和局限性 - 面向信仰编程 · /kubernetes-limitations

### DISCUSSION THREAD

**Discussion by draveness** [draveness/blog-comments#647] [Announcements]
https://draveness.me/kuberentes-limitations/

**Comment by zcong1993** [comment]
对于小公司小体量，用起来还是很爽的，而且云厂商现在把这东西做的也很无脑，很多工程师不懂原理都用在生产环境了。对于大规模确实有很多问题需要解决。

**Comment by yuchanns** [comment]
原来左神前段时间看的超售资源相关的文章是因为k8s利用率。看完文章又涨见识了🐂

**Comment by draveness** [comment]
> @zcong1993
> 对于小公司小体量，用起来还是很爽的，而且云厂商现在把这东西做的也很无脑，很多工程师不懂原理都用在生产环境了。对于大规模确实有很多问题需要解决。

:+1:


> @yuchanns
> 原来左神前段时间看的超售资源相关的文章是因为k8s利用率。看完文章又涨见识了🐂

是的哈哈哈

**Comment by silverfoxs** [comment]
硬多租户下面 实力=实例？

---

*2021-04-19 UPDATES：已修复*

**Comment by snowolfhawk** [comment]
博主，你的图是通过什么画的？真的很不错

**Comment by BillZong** [comment]
感谢左神的分享。收益良多。

单个集群体量到了那么大的情况下，必然会存在各种通信、管理问题，还是得分而治之。
多集群管理更符合未来计算资源利用的发展趋势，所以一直也是各种兴趣小组的关注重点。

目前，就多租户问题，确实是个难啃的骨头。
毕竟是依托于 Linux 系统基础上实现，内核 `container` 都没有实现“硬”多租户隔离。

**Comment by draveness** [comment]
> @BillZong
> 感谢左神的分享。收益良多。
>
> 目前，就多租户问题，确实是个难啃的骨头。
> 毕竟是依托于 Linux 系统基础上实现，内核 `container` 都没有实现“硬”多租户隔离。

这部分需要轻量级虚拟化来解决，例如 Firecracker

**Comment by dashjay** [comment]
找资料的时候发现有篇在腾讯云上，地址 https://cloud.tencent.com/developer/article/1532917

**Comment by BillZong** [comment]
好的，感谢

**Comment by cd-seven** [comment]
问下博客中的示例代码是什么主题

**Comment by LeoYang90** [comment]
> @draveness
> > @BillZong
> > 感谢左神的分享。收益良多。
> >
> > 目前，就多租户问题，确实是个难啃的骨头。
> > 毕竟是依托于 Linux 系统基础上实现，内核 `container` 都没有实现“硬”多租户隔离。
> 
> 这部分需要轻量级虚拟化来解决，例如 Firecracker

也可以试试 kata，对容器化比较友好

**Comment by DreamSaddle** [comment]
你好，请问我可以使用你[博客](https://draveness.me)中的一些样式自己使用吗？觉得挺简洁的！

**Comment by onecer** [comment]
> @snowolfhawk
> 博主，你的图是通过什么画的？真的很不错

文末有！！！

**Comment by linxuyalun** [comment]
左神好久没更新了啊

**Comment by sylzd** [comment]
要有多硬才算硬多租户呢?这块的分级及相应程度的风险是怎样的呢？

**Comment by woofyzhao** [comment]
时间来到了2021年，多集群有了karmada方案(基于kubefed)，多租户有了virtual cluster方案，大厂内部应该有结合起来用的多集群多租户方案。 

隔离 I/O、网络 和 CPU 缓存等资源，这些在单机上的多用户似乎也没做到硬隔离。个人理解要做到完全硬隔离和兼顾利用率只能预留hard quota+抢占, 对离线场景可能相对比较友好。
